lib/index.ts

```diff
@@ -144,45 +145,161 @@ class LogService {
    * Log a content on debug level
    * @param contents all the contents you intend to log in console/file, each content can be of any type of data
    */
-  d(...contents: any[]) {
+  d(...contents: any[]): LogServiceChainExtensions {
     contents.forEach((content) => {
       this.log(LogLevel.debug, content);
     });
+    
+    const self = this;
+    const elastic = this.globalConfig.elasticSearch;
+    return {
+      send() {
+        contents.forEach((content) => {
+          self.sendLog(LogLevel.debug, content, true)
+        });  
+      },
+      call(callback) {
+          if ( typeof callback !== 'function')
+            return console.error('Invalid parameter provided');
+          callback({
+            contents: contents,
+            loglevel: LogLevel.warning
+          }, elastic);
+      },
+    }
   }
 
   /**
    * Log a content on info level
    * @param contents all the contents you intend to log in console/file, each content can be of any type of data
    */
-  i(...contents: any[]) {
+  i(...contents: any[]): LogServiceChainExtensions {
     contents.forEach((content) => {
       this.log(LogLevel.info, content);
     });
+    
+    const self = this;
+    const elastic = this.globalConfig.elasticSearch;
+    return {
+      send() {
+        contents.forEach((content) => {
+          self.sendLog(LogLevel.info, content, true)
+        });
+      },
+      call(callback) {
+          if ( typeof callback !== 'function')
+            return console.error('Invalid parameter provided');
+          callback({
+            contents: contents,
+            loglevel: LogLevel.warning
+          }, elastic);
+      },
+    }
   }
 
   /**
    * Log a content on warning level
    * @param contents all the contents you intend to log in console/file, each content can be of any type of data
    */
-  w(...contents: any[]) {
+  w(...contents: any[]): LogServiceChainExtensions {
     contents.forEach((content) => {
       this.log(LogLevel.warning, content);
     });
+  
+    const self = this;
+  
+    const elastic = this.globalConfig.elasticSearch;
+    return {
+      send() {
+        contents.forEach((content) => {
+          self.sendLog(LogLevel.warning, content, true)
+        });  
+      },
+      call(callback) {
+          if ( typeof callback !== 'function')
+            return console.error('Invalid parameter provided');
+          callback({
+            contents: contents,
+            loglevel: LogLevel.warning
+          }, elastic);
+      },
+    }
   }
 
   /**
    * Log a content on error level
    * @param contents all the contents you intend to log in console/file, each content can be of any type of data
    */
-  e(...contents: any[]) {
+  e(...contents: any[]): LogServiceChainExtensions {
     contents.forEach((content) => {
       this.log(LogLevel.error, content);
     });
+    
+    const self = this;
+    const elastic = this.globalConfig.elasticSearch;
```

Comment:
Code duplicated . We can re-use the same code.

Reply:
The log service should call and extension function and dont need to know about elastic search